### ISSUE TITLE
Notion Template Sync adjustment

### ISSUE THREAD

**Issue by Alzx-Hzrz** [notion-enhancer/desktop#610]
TL;DR: *having the option to automatically desync templates once they are copied from a template button, in order to allow for master templates*

Notion's template system is incredibly useful but has no support for master templates

why is this feature necessary? how will it help? what existing shortcomings does it address?

I recently created a bunch of nested templates to organise my university notes, but I found no way to change one template and sync it to the rest without a bunch of extra steps

how will this feature appear/act?

I imagine this working essentially like a crossover between notion's template and sync blocks. 
you are able to sync templates at the moment, but when the template is then copied/produced, this will also be a synced template. In it's simplest form, a simple toggle could be implemented to unsync templates once they are duplicated.

**cons**

this may have a performance impact if used extensively, however that wouldn't be as a result of the feature itself, just the possible overuse of sync and template blocks.

**alternatives**

a similar solution would be to have all templates automatically stored in a "master template" page or file, and then when you create a template block, you get the option to choose from one of the master templates or make a new one. 
I guess this would be simpler for the end user but more work to implement?

does this need to be implemented by the enhancer? could an external service work better?

I haven't been able to find a "seamless" solution to this yet, and it seems like it would fir perfectly in notion enhancer!

Thank you!
![Notion_Master_Template_Issue](https://user-images.githubusercontent.com/34029068/136660983-c63d5342-ae14-44d2-b3b3-075d39399f94.png)

**Comment by Alzx-Hzrz** [comment]
Yeah I totally get what you're saying.

Please excuse my ignorance with this sort of thing because I have absolutely no experience, and enhancer seemed like the closest thing I could think of to try and implement a solution.

You said it can't interfere with actions but can create shortcuts? So (and I may be totally wrong with this), a shortcut applied to a template block that essentially activates the "new template" button and then activates the "desync" option on the template created (as they are both existing features?)

This now seems like a very specific request so I doubt it's practical or as simple as I thought :/

Even if it is possible.. I don't want to be the reason you spend hours trying to get it work if you don't see the benefit, It's just an idea!

On the side though, a massive thank you for the work you do on enhancer... it really is such a good addition to notion! (Now I can't go back!)


Is there anyone else that you know of (other than notion themselves) that works on similar solutions to my proposal? Or is it just something that notion doesn't offer to any extent past appearance?

Thanks for your quick response and help with this! Keep up the amazing work!

**Comment by l782993610** [comment]
In principle, Notion-enhancer executes JS scripts and attaches CSS styles. But the internal content of the user database cannot be changed. If you want to achieve such a complex function, you may need to manually construct a software back-end, customize the database form and use JS interaction. This amount of development is unacceptable.

**Comment by dragonwocky** [comment]
alright, so I've done some testing of this, and it could *potentially* be implemented by simulating a user click into block context menu and pressing "unsync block". however, that would need some sort of trigger to activate it, which... there isn't really. the options are:

a) the enhancer will unsync all synced blocks it encounters if there is a template following them on the same page - this could work, but could also cause some unwanted behaviour
b) add a keyboard shortcut to unsync the currently focused block

it's possible, but it is a fairly specialised use-case. I'm marking this for potential later implementation, so if a few other people would also use this and I've got other higher priority work out of the way i'll give it a go.

**Comment by dragonwocky** [comment]
sidenote @l782993610 you might be interested to know the next version of the enhancer will ship with high-level access to notion's database (more capable but also more difficult to use than the official API) so extensions can do things with that, but there will be heavy restrictions on it as it's possible to corrupt or lose content by using it.

**Comment by dragonwocky** [comment]
too highly specific of a feature, not something anyone could use, out of scope
